### PR TITLE
If AT was obtained by MRRT then don't check regular RT to refresh token

### PR DESCRIPTION
**Pull request by imakhalova**
Here is scenario: you sign in and put refresh token for 2 different resources into cache.
 Then you hit everyday 1 resource and put its refresh token into cache (including multiresource token).
 Then suddenly in a week you need to hit 2nd resource -> you go to cache -> take token for that resource -> hit server and it says that token expired -> you return error that you need to sign in again (launch UI).
 But we actually should try multiresource token if access token was obtained by MRRT or we don't have any access token for particular resource.

### CODE REVIEW

File: src/src/com/microsoft/aad/adal/AuthenticationContext.java [hunk 1]
```diff
@@ -1430,7 +1430,8 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
 
         // Trying to find refresh token first, if existed, will try to use the refresh token. 
         Logger.v(TAG, "Checking refresh tokens");
-        RefreshItem refreshItem = getRefreshToken(request);
+        boolean useMultiResourceRefreshToken = cachedItem == null ? true : cachedItem.getIsMultiResourceRefreshToken();
```
Comment (weijjia) [inline src/src/com/microsoft/aad/adal/AuthenticationContext.java]: If you're not going to change the value, make it final whenever possible

Comment (imakhalova) [inline src/src/com/microsoft/aad/adal/AuthenticationContext.java]: sure, let's make it final

Comment (agamemnum) [inline src/src/com/microsoft/aad/adal/AuthenticationContext.java]: nit: boolean useMultiResourceRefreshToken = cachedItem == null || cachedItem.getIsMultiResourceRefreshToken();

**Comment by weijjia** [comment]
left minor comment. I'm fine with the cache lookup, :shipit: if @RandalliLama is fine with it. :)

**Comment by RandalliLama** [comment]
I'm looking at this.  Will need to sign-off tomorrow morning.

**Comment by RandalliLama** [comment]
It seems factored weird in that I would expect getRefreshToken handle this since it is that methods job to find a refresh token regardless of what kind it is.  However, I won't block on that for now since it doesn't make the code significantly worse than it already is.

:shipit:

**Comment by imakhalova** [comment]
It's actually good feedback, and let's make it better. 
What if getRefreshToken will check RT first and if it multiresource take MRRT from cache?

**Comment by RandalliLama** [comment]
@imakhalova Exactly right.  I think that can wait for a better refactor however.  I we need a cache layer that handles all of this transparently to the rest of the code.  The cache logic should be in it's own class rather than spread across the various classes.

**Pull request merged**